### PR TITLE
More CI improvements

### PR DESCRIPTION
**Pull request by felixfontein**
CC @webknjaz

### CODE REVIEW

**Comment by felixfontein** [comment]
CI failures are expected.

**Review by gundalow — COMMENTED** [review]
Thanks for this
1) We can also remove the python matrix from `unit tests` as ansible-test as we are using the docker container which contains multiple Python versions.

**Comment by felixfontein** [comment]
True, will work on that later. I've already implemented this for some collection CIs. I think separating them by Python version mostly makes sense for larger collections where the unit tests per Python version take a longer time.

File: .github/workflows/ansible-test.yml [hunk 1]
```diff
@@ -109,7 +109,9 @@ jobs:
       # OPTIONAL If your unit test requires Python libraries from other collections
       # Install them like this
       - name: Install collection dependencies
-        run: ansible-galaxy collection install ansible.netcommon -p .
+        run: |
+          ansible-galaxy collection install ansible.netcommon -p .
+          ansible-galaxy collection install ansible.utils -p .
```
Comment (webknjaz) [inline .github/workflows/ansible-test.yml:114]: Does anything prevent us from adding the new package right into the first command invocation? It's usually better to feed everything into the same dependency resolution process.

**Comment by webknjaz** [comment]
@gundalow I usually prefer to keep the matrix because it's easier to maintain changes to it vs trying to find all of the places in the middle of the workflow needing updates

File: .github/workflows/ansible-test.yml (hunk 1 above)
Comment (felixfontein) [reply .github/workflows/ansible-test.yml:114]: I always forget that `ansible-galaxy collection install` actually allows for that :)

Comment (felixfontein) [reply .github/workflows/ansible-test.yml]: Fixed in 216b1e354411c4f4065d70d95a2118ffada489e0.

**Comment by felixfontein** [comment]
@webknjaz the main reason for not keeping the Python version there is that it is one less place where versions are kept and where ansible version/python version incompatibilities do not have to be explicitly listed, because ansible-test takes care of it.

**Comment by felixfontein** [comment]
@webknjaz check 4a05ece55432edc050d08690b7e3eee17bab525b, it actually makes maintenance easier since there's one less place where you have to manually update something.

**Comment by webknjaz** [comment]
But you still need to bump these versions once ansible-test stops supporting them... And you'll need to look for the places that need to be updated vs keeping the explicit setup in the header with all the other things

**Comment by felixfontein** [comment]
You do not mean any version modified in this PR, right?

**Review by gundalow — APPROVED** [review]
With the number of collections increasing, I don't want to have to maintain a list of Python versions to test against.

If (say) a Python library that is needed to support a Collection doesn't work on Python 2, then the GitHub Action for that specific collection can be updated to only use certain Python Versions.

**Pull request merged**